apps/users/models.py

```diff
@@ -145,6 +150,13 @@ def create_superuser(self, username, email, password):
         GroupUser.objects.create(user=user, group=admins)
         return user
 
+    def _generate_username(self, seed):
+        if self.model.objects.filter(username=seed).exists():
+            return '{seed}-{num}'.format(
+                seed=seed, num=random.randint(1000, 9999))
```

Comment:
I figured we'd either have to make it blow up at some point (3 attempts for example) or keep trying forever in which case you could DOS us by creating all of those users (`me@example.com-0000` through `me@example.com-9999`).

It seems pretty unlikely that this will fail so I opted for trying once.